### ISSUE TITLE
GT scattering flow

### ISSUE THREAD

**Issue by Jonsuff** [Jonsuff/YOLO_OD#19]
### GT box scattering flow (to pred shape)

```
Param : 
indexes = tf.TensorArray(tf.int32, size=1, dynamic_size=True)

updates = tf.TensorArray(tf.float32, size=1, dynamic_size=True)

y_true_out = tf.zeros(N, grid_size, grid_size, tf.shape(anchor_idxs)[0], 6)
```

1. y_true : (N, boxes, (y, x, h, w, class)) 

   -> 모든 anchor박스와 iou연산하여 가장 높은 anchor박스의 인덱스 구하기

   -> return : y_true (N, boxes, (y, x, h, w, class, best_anchor))

   

2. 모든 배치(N), 모든 박스(boxes)에 대해 for문 돌기 (N : i, boxes : j)

   

3. anchor_idxs : anchor mask([[6, 7, 8], [3, 4, 5], [0, 1, 2]]) 

   -> 현재 박스 하나(y_true\[i]\[j]\[best_anchor idx]와 1에서 구한 best anchor idx가 같은가?

   -> use tf.equal() -> return type = bool

4. 만약 같다면(tf.reduce_any(2번 결과)) : 

   ​		 box = y_true\[i]\[j]\[0:4]

​				anchor_idx = tf.cast(tf.where(anchor_eq, tf.int32))

​				grid_xy = tf.cast(box_xy // (1 / grid_size), tf.int32)

> box_xy는 전체에서 차지하는 비율로 되어있고, // 연산을 해줌으로써 소숫점을 생략할 수 있다.

​			indexes.write(idx, [i, grid_xy[1], grid_xy[0], anchor_idx\[0][0]])

​			updates.write(idx, box[0], box[1], box[2], box[3], 1, y_true\[i]\[j][4])

​			idx += 1

return tf.tensor_scatter_nd_update(y_true_out, indexes.stack(), updates.stack())

> y_true_out의 indexes에 해당하는 자리에 updates정보를 scatter한다.
